vimms/Controller/dia.py

```diff
@@ -51,35 +51,42 @@ def _process_scan(self, scan):
         scans = []
-        # make the SM1 scan
-        task = self.environment.get_default_scan_params(agc_target=self.ms1_agc_target,
-                                                        max_it=self.ms1_max_it,
-                                                        collision_energy=self.ms1_collision_energy,
-                                                        orbitrap_resolution=self.ms1_orbitrap_resolution)
-        task.set(ScanParameters.FIRST_MASS, self.min_mz)
-        task.set(ScanParameters.LAST_MASS, self.max_mz)
-        # task.set(ScanParameters.CURRENT_TOP_N, 10) # time sampling fix see iss18
-
-        scans.append(task)
-        self.scan_number += 1 # increase every time we make a scan
-        
-        # make the MS2 scan
-        dda_scan_params = ScanParameters()
-        dda_scan_params.set(ScanParameters.MS_LEVEL, 1)
-        isolation_windows = [[[self.min_mz,self.max_mz]]] # why tripple brackets needed?
-        dda_scan_params.set(ScanParameters.ISOLATION_WINDOWS, isolation_windows)
-        dda_scan_params.set(ScanParameters.COLLISION_ENERGY,self.ms2_collision_energy)
-        dda_scan_params.set(ScanParameters.AGC_TARGET,self.ms2_agc_target)
-        dda_scan_params.set(ScanParameters.MAX_IT,self.ms2_max_it)
-        dda_scan_params.set(ScanParameters.ORBITRAP_RESOLUTION,self.ms2_orbitrap_resolution)
-        dda_scan_params.set(ScanParameters.PRECURSOR_MZ, 0.5*(self.max_mz + self.min_mz))
-        dda_scan_params.set(ScanParameters.FIRST_MASS, self.min_mz)
-        dda_scan_params.set(ScanParameters.LAST_MASS, self.max_mz)
-        # dda_scan_params.set(ScanParameters.CURRENT_TOP_N,10) # time sampling fix see iss18
-
-
-        scans.append(dda_scan_params)
-        self.next_processed_scan_id = self.scan_number
-        self.scan_number += 1
+        if self.scan_to_process is not None:
+            
+            # make the MS2 scan
+            dda_scan_params = ScanParameters()
+            dda_scan_params.set(ScanParameters.MS_LEVEL, 2)
+            isolation_windows = [[[self.min_mz,self.max_mz]]] # why tripple brackets needed?
+            dda_scan_params.set(ScanParameters.ISOLATION_WINDOWS, isolation_windows)
+            dda_scan_params.set(ScanParameters.COLLISION_ENERGY,self.ms2_collision_energy)
+            dda_scan_params.set(ScanParameters.AGC_TARGET,self.ms2_agc_target)
+            dda_scan_params.set(ScanParameters.MAX_IT,self.ms2_max_it)
+            dda_scan_params.set(ScanParameters.ORBITRAP_RESOLUTION,self.ms2_orbitrap_resolution)
+            dda_scan_params.set(ScanParameters.PRECURSOR_MZ, 0.5*(self.max_mz + self.min_mz))
+            dda_scan_params.set(ScanParameters.FIRST_MASS, self.min_mz)
+            dda_scan_params.set(ScanParameters.LAST_MASS, self.max_mz)
+            # dda_scan_params.set(ScanParameters.CURRENT_TOP_N,10) # time sampling fix see iss18
+
+
+            scans.append(dda_scan_params)
+            #self.scan_number += 1
```

Comment:
Can you explain why you've removed this? It should be increased every time a scan is made...

Reply:
I removed code as it is no longer being used to update self.next_processed_scan_id. (if keep using self.scan_number to update self.next_processed_scan_id, it seems that self.next_processed_scan_id is being overwritten & reset back to initial value. For it has been incremented in initial stage while without scan.num_peaks by following code.
https://github.com/sdrogers/vimms/blob/7ceccd6d5e77464c12cd1c8624468144ba0007b4/vimms/Controller/topN.py#L48-L49
Agree to keep self.scan_number increment code that object property is valid .